### DISCUSSION TITLE
warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?

### DISCUSSION THREAD

**Discussion by SeeJayEmm** [requarks/wiki#7054] [Error / Bug Report]
Wiki.js in Docker, v2. It's fairly small and using the sqlite backend.

I am trying to re-point my Wiki.js to a new git repo but am having no success. I intended to start from scratch with the new repo so I purged the local repo.  When I click "Add Untracked Changes" the web server becomes non-responsive and the following repeats in the logs:

```
2024-03-07T15:07:32.494Z [MASTER] info: (STORAGE/GIT) Adding all untracked content...
2024-03-07T15:08:32.498Z [MASTER] warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2024-03-07T15:09:10.927Z [MASTER] warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2024-03-07T15:11:10.930Z [MASTER] warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2024-03-07T15:13:10.930Z [MASTER] warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2024-03-07T15:15:10.931Z [MASTER] warn: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
2024-03-07T15:16:51.355Z [MASTER] warn: Error while getting analytics code:  Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?
```

The only recourse I seem to have at this point is to restart the container.  The wiki never exports to the repo directory and I'm unable to sync.

**Comment by stanthewizzard** [comment]
Same issue here while trying to dump on local storage

**Comment by jjbrunton** [comment]
Same issue here. I really need to be able to export this content, has anyone found a workaround?

**Reply by stanthewizzard** [reply]
there is a solution
settings
in config change 

```
# ---------------------------------------------------------------------
# Database Pool Options
# ---------------------------------------------------------------------
# Refer to https://github.com/vincit/tarn.js for all possible options

pool:
   min: 2
   max: 10
```

uncomment min and max

No more issues in my case

**Reply by G-Pereira** [reply]
I am using postgresql does this apply as well? Which file is this?

**Reply by NGPixel** [reply]
Your config.yml file

This applies to all database options except SQLite.

**Reply by stanthewizzard** [reply]
sorry for not answering. Yes in config.yml

**Reply by BananaAcid** [reply]
how do I do this in the docker container? (requarks/wiki -> v2.5.305)

**Reply by blehz-be** [reply]
> Your config.yml file
> 
> This applies to all database options except SQLite.

I am running on SQLite and this definitely also applies.

**Reply by white-gecko** [reply]
@SeeJayEmm could you mark [@stanthewizzard 's answer](https://github.com/requarks/wiki/discussions/7054#discussioncomment-9263326) as resolution, if it also helped for you?

Would this make sense in the FAQ?

**Comment by jjbrunton** [comment]
Perfect, solved it for me - thank you!

**Reply by stanthewizzard** [reply]
welcome

**Comment by BananaAcid** [comment]
This does not seem to be possible for the docker container (requarks/wiki -> v2.5.305), as far as I can tell.

Any advice to be able to fix it there?

**Reply by SeeJayEmm** [reply]
If your config file is mounted ([Example](https://docs.requarks.io/install/docker#alternative-mount-the-config-file)) you can edit it on the docker host.

If your config file is not mounted you'll have to edit it in the container, knowing that your changes will get blown away with each update. Something like: `docker exec -it wiki-js-container-name /usr/bin/vi /wiki/config.yml`

But I'd strongly recommend you follow the directions to mount your config file if you want the changes to persist.

**Reply by BananaAcid** [reply]
I can't seem to be able to edit it in the running container. No console attachment seems to work (using QNAP Container-Station or Portainer).

Will have to ask the system guys, if they grant me terminal access to the Synology and QNAP servers to be able to fix that. Seems like building a new container with this fix is the only future proof solution. (I doubt, the clients, where we deployed it onto their servers, would grand me access to their system on terminal level)

**Comment by officeboyz** [comment]
I have some problem like this when build form current  branch and using dockerfile nodejs 20 .
i have update config.yml :
```
Port: 3000
bindIP: 0.0.0.0
db:
  type: postgres
  host: '$(DB_HOST)' 
  port: '$(DB_PORT)' 
  user: '$(DB_USER)' 
  pass: '$(DB_PASS)'
  db: '$(DB_DB)' 
  ssl: 0
pool: 
   min: 2
   max: 10 
 ``` 
But problem still come : 
2025-03-18T04:33:06.061Z [MASTER] [31merror[39m: Database Connection Error: Knex: Timeout acquiring a connection. The pool is probably full. Are you missing a .transacting(trx) call?

**Reply by beetaa** [reply]
same for me. problem still there.

**Comment by Kimbaras** [comment]
I'm getting the same error as well and this happened after the container was powered off for a while.

I've restarted the container but the error described showed up and blocked the correct usage.
Only me it's using (at least trying to) the instance so the load is basically non-existent.

The above instructions did not work for me.

@NGPixel any chance to have some feedback from you? 
This seems to be a recurring issue and I was unable to find anything usefull but this discussion.

For what is worth, running WikiJs 2.5.312.

Thanks!
